cloudflare/resource_cloudflare_zone.go

```diff
@@ -162,21 +171,34 @@ func resourceCloudflareZoneRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error finding Zone %q: %s", d.Id(), err)
 	}
 
+	// In the cases where the zone isn't completely setup yet, we need to
+	// check the `status` field and should it be pending, use the `Name`
+	// from `zone.PlanPending` instead to account for paid plans.
+	var plan string
+	if status, ok := d.GetOk("status"); ok {
+		if status == "pending" && zone.PlanPending.Name != "" {
+			plan = zone.PlanPending.Name
```

Comment:
This is currently failing at https://github.com/terraform-providers/terraform-provider-cloudflare/blob/1a55f4f97636d5663f81ddb39f69d4f17042befa/cloudflare/resource_cloudflare_zone.go#L160 

For some reason, the second time `Read` is called, this is the struct.

```
PlanPending: (cloudflare.ZonePlan) {
  ZonePlanCommon: (cloudflare.ZonePlanCommon) {
   ID: (string) "",
   Name: (string) "",
   Price: (int) 0,
   Currency: (string) "",
   Frequency: (string) ""
  },
  IsSubscribed: (bool) false,
  CanSubscribe: (bool) false,
  LegacyID: (string) "",
  LegacyDiscount: (bool) false,
  ExternallyManaged: (bool) false
 }
```

I'll continue digging into this one.

Reply:
Looking a the `zone` on it's own is 👌 

```
 Plan: (cloudflare.ZonePlan) {
  ZonePlanCommon: (cloudflare.ZonePlanCommon) {
   ID: (string) (len=32) "0feeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
   Name: (string) (len=12) "Free Website",
   Price: (int) 0,
   Currency: (string) (len=3) "USD",
   Frequency: (string) ""
  },
  IsSubscribed: (bool) true,
  CanSubscribe: (bool) false,
  LegacyID: (string) (len=4) "free",
  LegacyDiscount: (bool) false,
  ExternallyManaged: (bool) false
 },
 PlanPending: (cloudflare.ZonePlan) {
  ZonePlanCommon: (cloudflare.ZonePlanCommon) {
   ID: (string) (len=32) "94f3b7b768b0458b56d2cac4fe5ec0f9",
   Name: (string) (len=18) "Enterprise Website",
   Price: (int) 0,
   Currency: (string) (len=3) "USD",
   Frequency: (string) (len=7) "monthly"
  },
  IsSubscribed: (bool) false,
  CanSubscribe: (bool) true,
  LegacyID: (string) (len=10) "enterprise",
  LegacyDiscount: (bool) false,
  ExternallyManaged: (bool) true
 },
 Status: (string) (len=7) "pending",
 ... 
```

However using `zone.PlanPending` directly looks like an empty struct.

```
 PlanPending: (cloudflare.ZonePlan) {
  ZonePlanCommon: (cloudflare.ZonePlanCommon) {
   ID: (string) "",
   Name: (string) "",
   Price: (int) 0,
   Currency: (string) "",
   Frequency: (string) ""
  },
  IsSubscribed: (bool) false,
  CanSubscribe: (bool) false,
  LegacyID: (string) "",
  LegacyDiscount: (bool) false,
  ExternallyManaged: (bool) false
 },
```

🤔

Reply:
Have taken this up with Cloudflare Engineers as there is some stuff that isn't working in the API as expected.

Reply:
Looks like we'll need to swap to using the [zone subscription API](https://api.cloudflare.com/#zone-subscription-update-zone-subscription) to get this working as expected.

Reply:
Addressing the underlying issue in cloudflare/cloudflare-go#362 and once that lands, we will be good to continue shipping this.